### PR TITLE
Allow usage of v3 releases of zend-eventmanager and zend-stdlib

### PR DESCRIPTION
**Pull request by weierophinney**
This patch replaces #42.

This patch updates the dependency constraints as follows:

- PHP allows `^5.5 || ^7.0`, putting it in line with other components. That said, I'm aware of PHP 7 differences that currently mean tests do not pass (due to changes in sorting behavior).
- zend-eventmanager allows `^2.6 || ^3.0`
  - Provides slight updates to `Zend\Ldap\Node` to ensure that a lazy-loaded `EventManager` instance is created in a way that works against both major versions.
- zend-stdlib allows `^2.7 || ^3.0`

To ensure that the component works against each set of releases, this patch also updates the travis configuration to test against lowest supported verions, a lockfile, and latest versions.

### CODE REVIEW

**Comment by heiglandreas** [comment]
Thanks for the work! This makes handling the failing test to a top priority. That might delay tagging the 2.7 release but I believe that should be fixed before we get this into a new release!

**Comment by weierophinney** [comment]
@heiglandreas My pleasure. Good luck with that test. :smile:

File: src/Node.php [hunk 1]
```diff
@@ -123,8 +123,11 @@ public function __wakeup()
     public function getLdap()
     {
         if ($this->ldap === null) {
-            throw new Exception\LdapException(null, 'No LDAP connection specified.',
-                Exception\LdapException::LDAP_OTHER);
+            throw new Exception\LdapException(
+                null,
+                'No LDAP connection specified.',
+                Exception\LdapException::LDAP_OTHER
+            );
```
Comment (weierophinney) [inline src/Node.php]: These fixes were because phpcs was giving me grief about multiple arguments. I figured I'd go ahead and fix them.

Comment (heiglandreas) [inline src/Node.php]: I already thought so and it definitely improves readability!

File: test/bootstrap.php [hunk 2]
```diff
@@ -10,25 +10,13 @@
...
-if (class_exists('PHPUnit_Runner_Version', true)) {
-    $phpUnitVersion = PHPUnit_Runner_Version::id();
-    if ('@package_version@' !== $phpUnitVersion && version_compare($phpUnitVersion, '4.0.0', '<')) {
-        echo 'This version of PHPUnit (' . PHPUnit_Runner_Version::id() . ') is not supported'
-           . ' in the zend-ldap unit tests. Supported is version 4.0.0 or higher.'
-           . ' See also the CONTRIBUTING.md file in the component root.' . PHP_EOL;
-        exit(1);
-    }
-    unset($phpUnitVersion);
-}
```
Comment (weierophinney) [inline test/bootstrap.php]: Removed the above, as there was a series of PHPUnit releases using a broken sebastian/version that was reporting 4.X releases as 2.X, and thus making the above fail. There's no real reason for us to restrict running tests based on the version.

**Pull request merged**